front/components/vaults/VaultDataSourceViewContentList.tsx

```diff
@@ -98,27 +132,99 @@ export const VaultDataSourceViewContentList = ({
             : ""
         )}
       >
+        <DropdownMenu>
```

Comment:
This should be a component since we need to adapt its behavior based on the users' role.